teps/0004-tekton-integrations-annotations-statuses.md

```diff
@@ -0,0 +1,689 @@
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above.
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+#### Integration Event Handling
+
+A key aspect of the proposal is that user Pipelines are not directly responsible
+for handling integration events, and mainly focuses on how integrations can
+configure itself and provide information back to users within the Run. It
+intentionally does not define how integrations should handle events.
+
+It is up to individual integrations to decide how and when to handle integration
+logic. Handlers should be installed independently of the core Pipeline
+installation to allow users to customize their installation.
+
+As a recommendation, integrations should rely on some form of reconciler in
+order to reliably process events, but it's also fine to rely on a push
+mechanism, or even rely on PipelineTasks to handle and annotate integration
+status data.
+
+#### Annotation-less integrations
+
+Defining annotations is an option, not a requirement. If you want an integration
+to apply to all Runs with no configuration needed (or provided by another source
+like a ConfigMap), that is okay. Integrations of this type will still able to
+use status fields.
+
+This is useful for use cases like webhook/pubsub notifications for all Run
+events.
+
+#### Triggers
+
+While this proposal mainly focuses on integrations with Pipeline and Task Run
+resources, Triggers are another important aspect to handle incoming integration
+events.
+
+To start, we can treat triggers the same as normal user requests - users will
+need to set relevant annotations in the created Task/PipelineRun. This is
+similar to the existing user experience today of needing to configure
+integration Tasks in their pipeline.
+
+There are Trigger related features being discussed that would lead to
+incremental improvements:
+
+- ObjectMeta Templating (e.g. be able to template annotations on created objects
+  per trigger) ([#618](https://github.com/tektoncd/triggers/issues/618))
+- Integration Specific Interceptors / Bindings / Templates
+  ([#504](https://github.com/tektoncd/triggers/issues/504),
+  [#584](https://github.com/tektoncd/triggers/issues/584))
+
+While these are out of scope for this specific proposal, this proposal remains
+compatible with future changes to make configuring Triggers easier, since it is
+agnostic to the source of configuration.
+
+#### Notifications
+
+This relates heavily to existing work around
+[notifications](https://github.com/tektoncd/pipeline/issues/1740), and aims to
+build on it to support metadata for arbitrary integrations alongside the
+existing
+[Cloud Event notifications](https://github.com/tektoncd/pipeline/issues/2082).
+While this proposal is largely compatible with existing work, one change would
+be recommended - notably migrating `CloudEventDelivery` as an integration
+status.
```

Comment:
How does this relate to the current way the pipeline controller works, aka (if configured) sending cloud event to a given sink ? That feature doesn't write any `CloudEventDelivery` status, it's only when using the `CloudEvent` PipelineResource that we have this. (@afrittoli correct me if I'm wrong).